src/main/java/org/opensearch/knn/plugin/stats/KNNStatsConfig.java

```diff
@@ -85,5 +86,8 @@ public class KNNStatsConfig {
             StatNames.TRAINING_MEMORY_USAGE_PERCENTAGE.getName(),
             new KNNStat<>(false, new NativeMemoryCacheManagerSupplier<>(NativeMemoryCacheManager::getTrainingSizeAsPercentage))
         )
+        .put(StatNames.FAISS_FIELD.getName(), new KNNStat<>(true, new FieldWithEngineSupplier(KNNFlag.BUILT_WITH_FAISS)))
```

Comment:
If shard isnt present on node, I dont think this will get set. So, it will indicate if the node that is fielding the request has a shard that uses the given field. So cluster stat might be misleading.

Reply:
Yes, that's valid point.
If we do change it to node level stat, we will need to do an extra work on aggregating the per-node stats, which in this case is by using OR operator. I think it's equivalent of what we it does as a cluster level stat, isn't it? If that's no the case I agree, we need to change it to a node level

Reply:
I agree it should be a cluster level stat, but as it stands right now, we dont aggregate at the cluster level stats. The node that is handling the request just gets that information on its own. For instance, for index status, the node handling the request will just look up the index status. Aggregation would need to be implemented.

Reply:
do you mean that we do have standard place for implementing cluster level aggregation, or it's done outside the plugin and we simply define the node level stat for the field_by_engine?

Reply:
All cluster stats are retrieved here: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/plugin/transport/KNNStatsTransportAction.java#L79

Reply:
Now I see what you mean, that's a miss on our stats API design. I think aggregation needs to be implemented separately as it's require some re-design for the whole API. 
For this stat I think let's make it node level for now, in this case the info will be accurate. We can review this later once proper aggregation design is done

Reply:
Right. The stats API is definitely due for a refactor. 

I think keeping a node level stat for now makes sense. In this case, I think we can just add a "lucene_initialized" stat instead of 3 new node stats. It can be retrieved the same way as "lucene_field_exists". That way, we dont end up adding 3 new stats that function similarly to the initialized stats. What do you think @martin-gaievski?

Reply:
yes, seems that's a good idea, essentially new flag indicates same as existing "initialized" stat. Let me do the change

Reply:
> Right. The stats API is definitely due for a refactor.
> 
> I think keeping a node level stat for now makes sense. In this case, I think we can just add a "lucene_initialized" stat instead of 3 new node stats. It can be retrieved the same way as "lucene_field_exists". That way, we dont end up adding 3 new stats that function similarly to the initialized stats. What do you think @martin-gaievski?

Does that mean we will not have node level stat for nmslib engine and faiss engine?

Reply:
Not exactly, we do have it for nmslib and faiss today, and for lucene it will be in identical fashion. This is what the stats are with the recent commit:

```
{
    "_nodes":
    {
        "total": 1,
        "successful": 1,
        "failed": 0
    },
    "cluster_name": "integTest",
    "circuit_breaker_triggered": false,
    "model_index_status": null,
    "nodes":
    {
        "dWOq6KsVRWWJt938CfRvIg":
        {
            "graph_memory_usage_percentage": 0.0,
            "graph_query_requests": 0,
            "graph_memory_usage": 0,
            "cache_capacity_reached": false,
            "load_success_count": 0,
            "training_memory_usage": 0,
            "indices_in_cache":
            {},
            "script_query_errors": 0,
            "hit_count": 0,
            "knn_query_requests": 0,
            "total_load_time": 0,
            "miss_count": 0,
            "training_memory_usage_percentage": 0.0,
            "lucene_initialized": true,
            "graph_index_requests": 0,
            "faiss_initialized": false,
            "load_exception_count": 0,
            "training_errors": 0,
            "eviction_count": 0,
            "nmslib_initialized": false,
            "script_compilations": 0,
            "script_query_requests": 0,
            "graph_query_errors": 0,
            "indexing_from_model_degraded": false,
            "graph_index_errors": 0,
            "training_requests": 0,
            "script_compilation_errors": 0
        }
    }
}
```

Reply:
the "lucene_initialized" approach can be implemented with existing library initialized provider, this simplifies the code a lot. @jmazanec15, please take a look at recent commit

Reply:
For reference - I've created an issue https://github.com/opensearch-project/k-NN/issues/541 to track future work for stats aggregation